syntax = "proto3";
import "google/protobuf/timestamp.proto";

message Disconnected { string client_name = 1; }

message Coordinates {
  float latitude = 1;
  float longitude = 2;
  float altitude = 3;
  google.protobuf.Timestamp time = 4;
}

message MiniCallHome {
  string mac_address = 1;
  uint32 local_ip = 2;
  float cpu_temperature = 3;
  uint32 freq = 4;
  uint32 hw_model = 5;
  float volts = 6;
  int32 signal_dbm = 7;
  uint32 cellid = 8;
  uint32 network_type = 9;
  string codes = 10;
  uint64 totaldatarx = 11;
  uint64 totaldatatx = 12;
  google.protobuf.Timestamp time = 13;
}

message Status {
  oneof msg {
    Disconnected disconnected = 1;
    MiniCallHome mini_call_home = 3;
  }
}
